### PR TITLE
Process data

### PR DESCRIPTION
**Pull request by Fahim-Khan-P**
### **Added the:**

- [ ] Create models
- [ ] Create model classes for all entities as shown in the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png).
- [ ] Set up associations between models.

- Remember that author_id column in posts table should be the foreign_key for the users table.
- Use models to insert data

- [ ] Familiarize yourself with ['rails c' command](https://guides.rubyonrails.org/command_line.html#bin-rails-console).
- [ ] Open rails console for your project.
- [ ] Create at least one user by running the following code:

 ```
  first_user = User.create(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.')
   second_user = User.create(name: 'Lilly', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Poland.')
```

- [ ] Create at least 4 posts written by one of the users you created by running the following code:

 `  first_post = Post.create(author: first_user, title: 'Hello', text: 'This is my first post')`

- [ ] Create at least 6 posts comments for one of the posts you created by running the following code:

  ` Comment.create(post: first_post, author: second_user, text: 'Hi Tom!' )`

- [ ] Use other [CRUD methods](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/active_record_crud.md) to find, update and delete entities.

### Create custom methods

- [ ] Your models should include some additional methods.

 **Users**

1. A method that returns the 3 most recent posts for a given user.

 **Posts**

1. A method that updates the posts counter for a user.
2. A method which returns the 5 most recent comments for a given post.

 **Comments**

1. A method that updates the comments counter for a post.

**Likes**

1. A method that updates the likes counter for a post.
2. Go to rails c and check if your methods are working.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  has_many :likes
+
+  def update_posts_counter
+    user.posts_counter += 1
+  end
+
+  def recent_comments
+    comments.all.order(created_at: :desc).limit(5)
+  end
+end
```
Comment (Whoistolu) [inline app/models/post.rb:13]: - [ ] [OPTIONAL] I would like to also suggest that you can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_counter` method every time a new post has been saved to database, then make the `update_post_counter` method private.

```ruby
after_save :update_post_counter

  private

  def update_post_counter
   Code here ...
  end
```

File: app/models/comment.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  def update_comments_counter
+    post.comments_counter += 1
+  end
+end
```
Comment (Whoistolu) [inline app/models/comment.rb:8]: - [ ] [OPTIONAL] As I have recommended above, I would like to again suggest that you can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_comments_counter` method every time a new post has been saved to database, then make the `update_post_counter` method private.

```ruby
after_save :update_comments_counter

  private

  def increment_comment_counter
   Code here ...
  end
```

File: app/models/like.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class Like < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  def update_likes_counter
+    post.likes_counter += 1
+  end
+end
```
Comment (Whoistolu) [inline app/models/like.rb:8]: - [ ] [OPTIONAL] As I have recommended above, I would like to again suggest that you can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_likes_counter` method every time a new post has been saved to database, then make the `update_post_counter` method private.

```ruby
after_save :update_likes_counter

  private

  def increment_like_counter
   Code here ...
  end
```

**Review by Whoistolu — APPROVED** [review]
### Hi @Fahim-Khan-P 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

 - [ ]  _Check the inline comments under the review for sections to improve._

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/post.rb (hunk 1 above)
Comment (Fahim-Khan-P) [reply app/models/post.rb:13]: **Thank you very much @Whoistolu I will add the recommended codes.**

File: app/models/comment.rb (hunk 2 above)
Comment (Fahim-Khan-P) [reply app/models/comment.rb:8]: Thank you very much for your beneficial guide @Whoistolu I will add the recommended codes.

File: app/models/like.rb (hunk 3 above)
Comment (Fahim-Khan-P) [reply app/models/like.rb:8]: **Thank you Mr @Whoistolu for giving your time, I will add your suggestions.**

**Pull request merged**